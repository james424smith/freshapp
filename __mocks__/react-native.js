import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
