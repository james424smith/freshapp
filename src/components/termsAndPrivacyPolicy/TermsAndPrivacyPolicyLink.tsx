import React from "react";
import { View, TouchableOpacity } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/AuthStyles";
import {
  PRIVACY_POLICY_ROUTE,
  TERMS_AND_CONDITIONS_ROUTE,
} from "../../constants/routes";

type Props = { navigate: (arg: string) => void };

const TermsConditionsPrivacyPolicyLinks = (props: Props) => {
  const { navigate } = props;
  return (
    <View style={styles.conditionsPrivacyPolicyLinks}>
      <View style={styles.privacyPolicyView}>
        <Text style={styles.registerText}>By registering you accept our </Text>
        <TouchableOpacity
          testID={"privacy-policy-button"}
          onPress={() => navigate(PRIVACY_POLICY_ROUTE)}
        >
          <Text style={styles.privacyPolicyLink}>Privacy Policy</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.termsConditionsView}>
        <Text style={styles.andBetweenStyle}> and </Text>
        <TouchableOpacity
          testID={"terms-and-condition-button"}
          onPress={() => navigate(TERMS_AND_CONDITIONS_ROUTE)}
        >
          <Text style={styles.termsConditionsLink}>Terms & Conditions</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default TermsConditionsPrivacyPolicyLinks;
