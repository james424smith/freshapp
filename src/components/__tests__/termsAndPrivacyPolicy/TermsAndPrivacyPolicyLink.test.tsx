import React from "react";
import TermsAndPrivacyPolicyLink from "../../termsAndPrivacyPolicy/TermsAndPrivacyPolicyLink";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import {
  PRIVACY_POLICY_ROUTE,
  TERMS_AND_CONDITIONS_ROUTE,
} from "../../../constants/routes";

describe("AppColorScheme snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui and press privacy-policy-button", () => {
    const navigate = jest.fn();
    const { toJSON, getByTestId } = render(
      <TermsAndPrivacyPolicyLink navigate={navigate} />
    );

    fireEvent.press(getByTestId("privacy-policy-button"));
    expect(navigate).toHaveBeenCalledWith(PRIVACY_POLICY_ROUTE);

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui and press terms-and-condition-button", () => {
    const navigate = jest.fn();
    const { toJSON, getByTestId } = render(
      <TermsAndPrivacyPolicyLink navigate={navigate} />
    );

    fireEvent.press(getByTestId("terms-and-condition-button"));
    expect(navigate).toHaveBeenCalledWith(TERMS_AND_CONDITIONS_ROUTE);

    expect(toJSON()).toMatchSnapshot();
  });
});
