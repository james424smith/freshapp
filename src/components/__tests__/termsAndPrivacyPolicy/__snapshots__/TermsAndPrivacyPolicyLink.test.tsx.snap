// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppColorScheme snapshot test should render correct ui and press privacy-policy-button 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#78849E",
            "fontSize": 22.5,
          },
        ]
      }
    >
      By registering you accept our 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="privacy-policy-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#78849E",
              "fontSize": 22.5,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Privacy Policy
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#78849E",
            "fontSize": 22.5,
          },
        ]
      }
    >
       and 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="terms-and-condition-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#78849E",
              "fontSize": 22.5,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Terms & Conditions
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppColorScheme snapshot test should render correct ui and press terms-and-condition-button 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#78849E",
            "fontSize": 22.5,
          },
        ]
      }
    >
      By registering you accept our 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="privacy-policy-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#78849E",
              "fontSize": 22.5,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Privacy Policy
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#78849E",
            "fontSize": 22.5,
          },
        ]
      }
    >
       and 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="terms-and-condition-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#78849E",
              "fontSize": 22.5,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Terms & Conditions
      </Text>
    </View>
  </View>
</View>
`;
